src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -455,7 +463,35 @@ public void invalidated(Observable observable) {
             }
         });
 
-        setCenter(splitPane);
+        setCenter(notificationPane);
+    }
+
+    // pass in the text value on the left, and the buttonName on the right of the notification bar
+    // and an eventHandler that act when the user click the button
+    public void showNotificationPane(String text, String buttonName, Consumer<ActionEvent> eventHandler) {
+        notificationPane.setText(text);
+        ObservableList<Action> actions = notificationPane.getActions();
+        Action action = new Action(buttonName, eventHandler);
+        // avoid multiple actions be added to the notification Pane, only one action is allowed
+        if (actions.isEmpty()) {
+            actions.add(action);
+        } else {
+            actions.set(0, action);
+        }
+        notificationPane.show();
+        // the notification bar will disappear if the user doesn't have actions in 10 seconds
+        new Thread(() -> {
```

Comment:
It's not really good to use the low-level Thread here. Most of the time it's better to use one of the more modern Conccurrency things, a good overview is here  https://www.baeldung.com/java-concurrency 

**For this specific case:**
 Better use a Timer.
 We have one example in GlobalSearchBar that you can look at, e.g.
`        Timer searchTask = FxTimer.create(java.time.Duration.ofMillis(SEARCH_DELAY), this::performSearch);
`